src/crypto/symmkey.cpp

```diff
@@ -2,24 +2,57 @@
 // Licensed under the Apache 2.0 License.
 #include "symmkey.h"
 
+#include "ds/logger.h"
 #include "error.h"
 
 #include <mbedtls/aes.h>
+#include <mbedtls/error.h>
 #include <mbedtls/gcm.h>
 
 namespace crypto
 {
+  inline std::string str_err(int err)
```

Comment:
This could also be `tls::error_string`. I don't think `crypto` should really depend on `tls`, but its fine for now - they're both slightly oddly named and probably ought to be merged.

Reply:
If I can do that without changing dependencies, I will. Otherwise the merge will go in another PR, this one's large enough.